src/NHibernate/Properties/FieldAccessor.cs

```diff
@@ -158,11 +160,13 @@ private string GetFieldName(string propertyName)
 		/// An <see cref="IGetter"/> that uses a Field instead of the Property <c>get</c>.
 		/// </summary>
 		[Serializable]
-		public sealed class FieldGetter : IGetter, IOptimizableGetter
+		public sealed class FieldGetter : IGetter, IOptimizableGetter, IDeserializationCallback
 		{
 			private readonly FieldInfo field;
 			private readonly System.Type clazz;
 			private readonly string name;
+			[NonSerialized]
+			private Lazy<Func<object, object>> _getDelegate;
```

Comment:
Using a lazy here or in property accessor looks like #1904 laziness, which was rejected: it may reduce start-up time but will cause warming-up more hard to achieve.

It looks even more debatable than #1904, because #1904 was more likely to allow not initializing things never actually used by the application. But about getter/setter used notably for accessing an entity identifier, having them never used would require the application to map useless entities, which should normally not be done.

Reply:
I've made an alternative approach (#1944) that avoids this problem.